crates/fuel-core/src/service/adapters/block_importer.rs

```diff
@@ -113,6 +115,10 @@ impl ImporterDatabase for Database {
     fn latest_block_height(&self) -> StorageResult<BlockHeight> {
         self.latest_height()
     }
+
+    fn update_tx_count(&self, new_txs_count: u64) -> StorageResult<u64> {
```

Comment:
Maybe `update_tx_count` should also be named `increase_tx_count`, and the inner call should be namespaced. The name `update_tx_count` suggests that we are setting the TX count to the argument we pass in, but the inner behaviour is that we are setting the TX count to the current TX count + the argument we pass in.